### ISSUE TITLE
Error: Unable to resolve module ./Libraries/Components/AccessibilityInfo/AccessibilityInfo

### ISSUE THREAD

**Issue by satyachaurasia** [facebook/react-native#37196]
### Description

Whenever I'm trying to run my app in my device through xcode I get 
Error: Unable to resolve module ./Libraries/Components/AccessibilityInfo/AccessibilityInfo from /Users/satyatest2/Work/mobileapp/MobileApp/node_modules/react-native/index.js: 

None of these files exist:
  * node_modules/react-native/Libraries/Components/AccessibilityInfo/AccessibilityInfo(.native|.native.js|.js|.native.json|.json|.native.ts|.ts|.native.tsx|.tsx)
  * node_modules/react-native/Libraries/Components/AccessibilityInfo/AccessibilityInfo/index(.native|.native.js|.js|.native.json|.json|.native.ts|.ts|.native.tsx|.tsx)
  12 |
  13 | // Components
> 14 | import typeof AccessibilityInfo from './Libraries/Components/AccessibilityInfo/AccessibilityInfo';
     |                                       ^
  15 | import typeof ActivityIndicator from './Libraries/Components/ActivityIndicator/ActivityIndicator';
  16 | import typeof Button from './Libraries/Components/Button';
  17 | import typeof DatePickerIOS from './Libraries/Components/DatePicker/DatePickerIOS';
    at ModuleResolver.resolveDependency (/Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:211:15)
    at DependencyGraph.resolveDependency (/Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/node-haste/DependencyGraph.js:413:43)
    at Object.resolve (/Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/lib/transformHelpers.js:317:42)
    at resolve (/Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/DeltaBundler/traverseDependencies.js:629:33)
    at /Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/DeltaBundler/traverseDependencies.js:645:26
    at Array.reduce (<anonymous>)
    at resolveDependencies (/Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/DeltaBundler/traverseDependencies.js:644:33)
    at /Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/DeltaBundler/traverseDependencies.js:329:33
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/satyatest2/Work/mobileapp/MobileApp/node_modules/metro/src/DeltaBundler/traverseDependencies.js:137:24)

I've checked the file is already there.

### React Native Version

0.64.4

### Output of `npx react-native info`

System:
    OS: macOS 13.3.1
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 264.35 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 14.8.0 - ~/.nvm/versions/node/v14.8.0/bin/node
    Yarn: Not Found
    npm: 6.14.7 - ~/.nvm/versions/node/v14.8.0/bin/npm
    Watchman: 2022.08.15.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.12.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 14.1/14B47b - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.15 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: ^0.64.4 => 0.64.4
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

I was not able to reproduce in other projects with the same package.json

### Snack, code example, screenshot, or link to a repository

![image](https://user-images.githubusercontent.com/40147144/235752632-6d9c2b47-0b8a-42f4-8790-c6341fe6bce8.png)

**Comment by satyachaurasia** [comment]
This ticket is duplicate of https://github.com/facebook/react-native/issues/36794

**Comment by lichenbuliren** [comment]
react-native version 0.63 meets this issue too
<img width="892" alt="image" src="https://github.com/facebook/react-native/assets/8489343/0ab97aaa-87f9-4b19-ab2c-ab7d62fdc466">

**Comment by Jack-QMY** [comment]
> react-native 版本 0.63 也遇到了这个问题 <img alt="图片" width="892" src="https://user-images.githubusercontent.com/8489343/241117953-0ab97aaa-87f9-4b19-ab2c-ab7d62fdc466.png">
你好，请问你解决了吗？我今天也遇到该问题

**Comment by mayursinhgohil** [comment]
react-native version 0.63.2, facing the same issue:

<img width="1279" alt="Screenshot 2023-06-18 at 7 39 06 PM" src="https://github.com/facebook/react-native/assets/29255184/cfa79524-5f70-4467-be46-139b90e773e2">

**Comment by mayursinhgohil** [comment]
So the issues seems to be with IOS 16.4, It’s only affecting development (so building and using in production should not cause any issues), and afaik there’s no way around it but it’s being looked at.

So on **lowering my simulator/device IOS version to 16.2, the error went away.**

**Comment by xjoshuaxchow** [comment]
react-native version 0.63.5, facing the same issue:

**Comment by mishkeTz** [comment]
I got the same error for 0.64.4
Seems like they fixed it only for newer versions of RN...

**Comment by jljonnard** [comment]
I experienced the exact same issue on Android in RN 0.72.4 and deleting the folder `cxx` in `Android/App` has worked for me.

**Comment by dangnguyen1004** [comment]
Having the same issue on 0.63.5. I have to rename any file with extension *.ios.js => *.js in the folder node_modules/react-native/Libraries/... but there so many file to rename

**Comment by dangnguyen1004** [comment]
Such a damage if I have to update RN for my project; it's stable now except for this supper annoying issue.

**Comment by eds9nsilva** [comment]
> react-native versão 0.63.2, enfrentando o mesmo problema:
> 
> <img alt="Captura de tela 18/06/2023 às 19h39 18h" width="1279" src="https://user-images.githubusercontent.com/29255184/246749688-cfa79524-5f70-4467-be46-139b90e773e2.png">

Did you find any way to resolve it?

**Comment by dangnguyen1004** [comment]
> > react-native versão 0.63.2, enfrentando o mesmo problema:
> > <img alt="Captura de tela 18/06/2023 às 19h39 18h" width="1279" src="https://user-images.githubusercontent.com/29255184/246749688-cfa79524-5f70-4467-be46-139b90e773e2.png">
> 
> Did you find any way to resolve it?

Poorly, it's no. I still got this when developing

**Comment by yogendrajs** [comment]
@react-native-bot or anyone from the team, please can you escalate this issue?

**Comment by satyachaurasia** [comment]
> > Having the same issue on 0.63.5. I have to rename any file with extension *.ios.js => *.js in the folder node_modules/react-native/Libraries/... but there so many file to rename
> 
> ```shell
> cd node_modules
> find . -name '*.ios.js' -exec sh -c 'mv {} $(dirname {})/$(basename {} .ios.js).js' \;
> ```

The solution is provided in this comment: https://github.com/facebook/react-native/issues/36794#issuecomment-1500880284
